### DISCUSSION TITLE
How to force re-update of a dependency?

### DISCUSSION THREAD

**Discussion by l0b0** [renovatebot/renovate#22477] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

14.5.0

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I want to force Renovate to create a *new* branch and merge request for an already up-to-date dependency. I need to do this to test a new [regex manager](https://gitlab.com/engmark/mypy-exercises/-/merge_requests/392/diffs?commit_id=35ee48aec0eb7ab74c4ffa5e630c9321ce0a0037) before merging it, both to ensure it's tested properly and to ensure I don't have to wait for the next version of the third party package for it to run.

### What you tried so far.

[This](https://gitlab.com/engmark/engmark.gitlab.io/-/merge_requests/263/diffs): Reverting the relevant Renovate commit on the branch with the `renovate.json` change, then running Renovate with `RENOVATE_BASE_BRANCHES` pointing to the aforementioned branch.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
I don't understand exactly what you mean, but maybe the new `renovate --platform=local --dry-run=extract` feature might help you test your regex?

**Reply by l0b0** [reply]
I don't think so. I already checked that the regex matches. I want to check that the Renovate configuration does exactly what I expect, in a realistic setting (actually creating a new branch and MR with the expected content for a dependency which needs updating) and with a realistic setup (GitLab CI, not my local machine). The link has more info, as I tried to document the process for future reference but got stuck on this specific point.

**Reply by rarkins** [reply, marked as answer]
I'm still not certain I understand what you mean, but I'm pretty sure it's not possible

**Reply by l0b0** [reply]
In that case, is there any way at all to test Renovate configuration changes properly? I couldn't find anything in the docs or Google.

**Reply by rarkins** [reply]
Fork or import and repo into a new repo, if you need a full replication

**Reply by l0b0** [reply]
OK, I guess that's a no. :fearful: Love Renovate otherwise, tho!
